src/js/editor/editor.js

```diff
@@ -648,7 +622,7 @@ class Editor {
 
   insertSectionAtCursor(newSection) {
     let newRenderNode = this._renderTree.buildRenderNode(newSection);
-    let renderNodes = this.getRenderNodesWithCursor();
+    let renderNodes = this.cursor.activeSections.map(s => s.renderNode);
```

Comment:
I think we are close to capturing enough of the edit actions to be able to ensure that renderNodes and postNodes stay in sync, and at that point it may make more sense to move more of the actions through renderNodes directly. The extra boilerplate around render nodes (build render node, attach it, mark it clean/dirty) I think adds a little friction to using them as freely, but that can be easily changed.